orders.go

```diff
@@ -123,8 +123,8 @@ func (repository orderRepository) Create(order shopify.Order) (shopify.Order, er
 // OrderDTO represents a Shopify order in HTTP requests and responses
 type OrderDTO struct {
 	BillingAddress           AddressDTO       `json:"billing_address,omitempty"`
-	ClosedAt                 time.Time        `json:"closed_at,omitempty"`
-	CreatedAt                time.Time        `json:"created_at,omitempty"`
+	ClosedAt                 *time.Time       `json:"closed_at,omitempty"`
```

Comment:
Will this have to be applied everywhere we have use `time.Time` in all our `DTOs` like products? May as well do them all in this PR if so?

Reply:
It will indeed... I wonder if we can wait on this because times tight now and its archived but that kinda does not matter at the moment. I dunno...

Reply:
yeah lets